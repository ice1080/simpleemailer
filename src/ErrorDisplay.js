import React, { Component } from 'react';

export default class ErrorDisplay extends Component {
  constructor(props) {
    super(props);

    this.state = {
      invalidEmails: [],
      emptyFields: [],
    };
  }

  updateErrors(emptyFields, invalidEmails) {
    this.setState({emptyFields: emptyFields, invalidEmails: invalidEmails});
  }

  renderInvalidEmailErrors() {
    if (this.state.invalidEmails.length > 0) {
      let i = 0;
      return (
        <React.Fragment>
          The following fields contain invalid email addresses:
          <ul>{
              this.state.invalidEmails.map(invalidEmailField => {
                i++;
                return (<li key={i} >{invalidEmailField}</li>);
              })
          }</ul>
          </React.Fragment>
      );
    } else {
      return '';
    }
  }

  renderEmptyFieldsErrors() {
    if (this.state.emptyFields.length > 0) {
      let i = 0;
      return (
        <React.Fragment>
          The following fields are missing information:
          <ul>{
              this.state.emptyFields.map(emptyField => {
                i++;
                return (<li key={i} >{emptyField}</li>);
              })
          }</ul>
        </React.Fragment>
      );
    } else {
      return '';
    }
  }

  render() {
    return(
      <React.Fragment>
        <div style={{color: 'red'}} >
          {this.renderInvalidEmailErrors()}
          {this.renderEmptyFieldsErrors()}
        </div>
        <br/>
      </React.Fragment>
    );
  }
}
