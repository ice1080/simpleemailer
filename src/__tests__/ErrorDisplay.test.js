import React from 'react';
import {shallow, mount} from 'enzyme';
import ErrorDisplay from '../ErrorDisplay';

describe('ErrorDisplay', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ErrorDisplay />);
  });

  test('default state', () => {
    expect(wrapper.state()).toMatchSnapshot();
  });

  test('updateErrors', () => {
    let emptyFields = ['a', 'b'];
    let invalidEmails = ['c', 'd'];

    wrapper.instance().updateErrors(emptyFields, invalidEmails);

    expect(wrapper.state()).toMatchSnapshot();
  });

  test('renderInvalidEmailErrors none', () => {
    expect(wrapper.text()).toBe('');
  });

  test('renderInvalidEmailErrors', () => {
    wrapper.setState({invalidEmails: ['error1', 'error2']});
    expect(wrapper.text()).toBe('The following fields contain invalid email addresses:error1error2');
  });

  test('renderEmptyFieldsErrors none', () => {
    expect(wrapper.text()).toBe('');
  });

  test('renderEmptyFieldsErrors', () => {
    wrapper.setState({emptyFields: ['errorA', 'errorB']});
    expect(wrapper.text()).toBe('The following fields are missing information:errorAerrorB');
  });
});
