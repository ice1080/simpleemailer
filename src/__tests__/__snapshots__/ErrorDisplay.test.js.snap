// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay default state 1`] = `
Object {
  "emptyFields": Array [],
  "invalidEmails": Array [],
}
`;

exports[`ErrorDisplay updateErrors 1`] = `
Object {
  "emptyFields": Array [
    "a",
    "b",
  ],
  "invalidEmails": Array [
    "c",
    "d",
  ],
}
`;
